### PR TITLE
Add model config to all watcher ModelInfo

### PR DESCRIPTION
**Pull request by wallyworld**
The allwatcher now supplies ModelInfo objects containing the model config as a name-value map.

A nasty bug was discovered and fixed whereby the models were cached twice and the wrong model entities where being updated when new config arrived. The issue didn't matter until we started storing additional data (config) with the watcher model entities.

QA: bootstrap and smoke test GUI

### CODE REVIEW

File: api/controller/legacy_test.go [hunk 1]
```diff
@@ -160,14 +161,28 @@ func (s *legacySuite) TestWatchAllModels(c *gc.C) {
 		c.Assert(deltas, gc.HasLen, 1)
 		modelInfo := deltas[0].Entity.(*multiwatcher.ModelInfo)
 
-		env, err := s.State.Model()
+		model, err := s.State.Model()
 		c.Assert(err, jc.ErrorIsNil)
+		cfg, err := model.Config()
+		c.Assert(err, jc.ErrorIsNil)
+
+		// Resource tags are marshalleded as map[string]interface{}
```
Comment (axw) [inline api/controller/legacy_test.go]: marshalled

File: state/allwatcher.go [hunk 2]
```diff
@@ -791,6 +798,16 @@ func (s *backingSettings) updated(st *State, store *multiwatcherStore, id string
 	case nil:
 		// The parent info doesn't exist. Ignore the status until it does.
 		return nil
+	case *multiwatcher.ModelInfo:
+		// We need to construct a model config so that coercion
+		// of raw settings values occurs.
+		cfg, err := config.New(config.UseDefaults, s.Settings)
```
Comment (axw) [inline state/allwatcher.go]: I think you should be using NoDefaults here, like in State.ModelConfig. Config should be complete when it's stored in state.

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**